templates/static/scripts.js

```diff
@@ -56,3 +64,30 @@ document.getElementById('url-form').addEventListener('submit', async function(ev
         imageResultDiv.textContent = `Error: ${error.message}`;
     }
 });
+
+async function downloadImage(imageUrl, text, headlineElement) {
+    const x = parseFloat(headlineElement.getAttribute('data-x')) || 0;
+    const y = parseFloat(headlineElement.getAttribute('data-y')) || 0;
+
+    const response = await fetch('/download-image', {
```

Comment:
Getting `406 Client Error: Not Acceptable for url: https://thinkingofbermuda.com/wp-content/uploads/2019/08/Traveler-Charters-4-480x340.jpg` when trying to download any image

Reply:
The `406 Not Acceptable` error typically occurs when the server cannot generate a response that is acceptable according to the `Accept` headers sent in the request. 

To address this issue, ensure that the `Accept` header in the request to download the image is set to accept any type of content. You can modify the fetch request in the `downloadImage` function to include the appropriate headers:

```javascript
const response = await fetch('/download-image', {
    method: 'POST',
    headers: {
        'Content-Type': 'application/json',
        'Accept': '*/*'  // Add this line to accept any content type
    },
    body: JSON.stringify({ image_url: imageUrl, text: text, x: x, y: y })
});
```

This should help resolve the `406 Not Acceptable` error.

Reply:
The `406 Not Acceptable` error typically occurs when the server cannot generate a response that is acceptable to the client based on the `Accept` headers sent in the request. 

To resolve this, ensure that the `Accept` header in the request to `/download-image` is set to accept `image/png`. You can add this header in the `fetch` request within the `downloadImage` function in your JavaScript code. 

Here's a brief example of how to set the `Accept` header:

```javascript
const response = await fetch('/download-image', {
    method: 'POST',
    headers: {
        'Content-Type': 'application/json',
        'Accept': 'image/png'
    },
    body: JSON.stringify({ image_url: imageUrl, text: text, x: x, y: y })
});
```

This should help the server understand that the client expects an image in PNG format.

Reply:
The `406 Client Error: Not Acceptable` typically occurs when the server cannot generate a response that matches the criteria specified in the request's `Accept` header. To resolve this, you can try setting the `Accept` header in the request to a more generic value, such as `*/*`.

Here's a quick fix for the `download_image` function in `app/routes.py`:

```python
headers = {
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3",
    "Accept": "*/*",
}
response = requests.get(request.image_url, headers=headers)
```

This should help the server understand that it can return any type of content.